src/streams/openssl.c

```diff
@@ -602,6 +600,8 @@ int openssl_connect(git_stream *stream)
 	if ((ret = SSL_connect(st->ssl)) <= 0)
 		return ssl_set_error(st->ssl, ret);
 
+	st->connected = true;
```

Comment:
Makes sense, too. One might wonder about whether the BIO will still get free'd if we never call `SSL_shutdown`. But documentation of `SSL_set_bio` states that BIO's ownership is transferred to the SSL handle and will get free'd with `BIO_free_all` as soon the SSL handle is free'd.